clients/fossid-webapp/src/main/kotlin/serviceExtensions.kt

```diff
@@ -0,0 +1,273 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+@file:Suppress("TooManyFunctions")
+
+package org.ossreviewtoolkit.fossid
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+
+import kotlin.reflect.KClass
+
+import org.ossreviewtoolkit.model.jsonMapper
+
+import retrofit2.Call
+
+private const val SCAN_GROUP = "scans"
+private const val PROJECT_GROUP = "projects"
+
+/**
+ * This function verifies that a request to FossID server was successful
+ * @param operation the label of the operation, for logging
+ * @param withDataCheck if true, the payload will be checked for existence
+ */
+fun EntityPostResponseBody<*>?.checkResponse(operation: String, withDataCheck: Boolean = true) {
+    checkNotNull(this)
+    require(error == null) {
+        "Could not '$operation'. Additional information : $error"
+    }
+    if (withDataCheck) {
+        requireNotNull(data) {
+            "No Payload received for '$operation'. Additional information : $error"
+        }
+    }
+}
+
+/**
+ * The list operations in FossID have inconsistent return types depending on the amount of entities returned.
+ * This function streamlines these entities to a list.
+ */
+fun <T : Any> EntityPostResponseBody<Any>.toList(cls: KClass<T>): List<T> {
+    // the list  operation returns different json depending on the amount of scans
+    return when (val data = data) {
+        is List<*> -> {
+            data.map { jsonMapper.convertValue(it, cls.java) }
+        }
+        is Map<*, *> -> {
+            data.values.map { jsonMapper.convertValue(it, cls.java) }
+        }
+        is Boolean -> {
+            emptyList()
+        }
+        else -> {
+            error("Cannot process the returned values")
+        }
+    }
+}
+
+/**
+ * Get a FossID project
+ * @param apiKey the key to query the API
+ * @param user the user querying the API
+ * @param projectCode the code of the project
+ */
+fun FossIdRestService.getProject(apiKey: String, user: String, projectCode: String) = getProject(
+        PostRequestBody("get_information", PROJECT_GROUP, apiKey, user, "project_code" to projectCode)
+)
```

Comment:
Wrong formatting. The wanted style would be

```
fun FossIdRestService.getProject(apiKey: String, user: String, projectCode: String) =
    getProject(
        PostRequestBody("get_information", PROJECT_GROUP, apiKey, user, "project_code" to projectCode)
    )
```

Similar below.

Reply:
done